File: app/src/main/java/cn/zjl/datacollector/data/entity/MeasurementPointEntity.java
package cn.zjl.datacollector.data.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "data_point")
public class MeasurementPointEntity {

    @PrimaryKey(autoGenerate = true)
    public long id;

    public float name;

    public int use;

    public String note;

    public int type;

    public long dataLineId;

    public int status;

    public boolean isQualified;

    public boolean isSynced;

    public long collectionTime;

    public double latitude;

    public double longitude;

    public double altitude;

    public String syncError;

    public long createdAt;

    public long updatedAt;
}

File: app/src/main/java/cn/zjl/datacollector/data/entity/MeasurementPointEntity.kt
package cn.zjl.datacollector.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "data_point")
data class MeasurementPointEntity(
    @PrimaryKey(autoGenerate = true) var id: Long = 0,
    var name: Float = 0f,
    var use: Int = 0,
    var note: String = "",
    var type: Int = 0,
    var dataLineId: Long = 0,
    var status: Int = 0,
    var isQualified: Boolean = false,
    var isSynced: Boolean = false,
    var collectionTime: Long = 0,
    var latitude: Double = 0.0,
    var longitude: Double = 0.0,
    var altitude: Double = 0.0,
    var syncError: String? = null,
    var createdAt: Long = 0,
    var updatedAt: Long = 0
)
